drivers/storage/portworx/util/util.go

```diff
@@ -1471,19 +1454,18 @@ func CountStorageNodes(
 			if !isDRSetup {
 				storageNodesCount++
 			} else {
-				if _, ok := k8sNodesStoragePodCouldRun[node.SchedulerNodeName]; ok {
```

Comment:
I just realized that we will still need to keep this check for pre-3.1.1 PX versions. Before 3.1.1, the enumerate response would not have the cluster_domain value populated.

So, we would end up counting all the nodes in both the domains and the upgrade will never proceed. Thus to avoid breaking the existing behavior for <3.1.1 DR clusters, let's calculate using the k8sNodeStoragePodCouldRun map, else we compare the cluster domains.

Also, ensure that all the nodes are on 3.1.1 before we can use the cluster domain field (just like how we do for quorum flag above).